app/src/main/java/com/android/shelfLife/model/user/UserRepositoryFirestore.kt

```diff
@@ -87,6 +87,48 @@ class UserRepositoryFirestore @Inject constructor(
         }
     }
 
+    override suspend fun getUserIds(userEmails: Set<String?>): Map<String, String> {
+        if (userEmails.isEmpty()) return emptyMap()
+
+        val emailToUserId = mutableMapOf<String, String>()
+        val chunks = userEmails.filterNotNull().chunked(10)
+
+        for (chunk in chunks) {
+            val query = db.collection("users")
+                .whereIn("email", chunk)
+                .get()
+                .await()
+            for (doc in query.documents) {
+                val email = doc.getString("email")
+                val userId = doc.id
+                if (email != null) emailToUserId[email] = userId
+            }
+        }
+
+        return emailToUserId
+    }
+
+    override suspend fun getUserEmails(userIds: List<String>): Map<String, String> {
+        if (userIds.isEmpty()) return emptyMap()
+
+        val uidToEmail = mutableMapOf<String, String>()
+        val chunks = userIds.chunked(10)
```

Comment:
Just for my understanding, are there any optimizations if we chunk the data? I understand that this is a small scale application but do we get significant performance if we chunk the data?

Reply:
No, the point is that the limit of firestore is 10 concurrent gets at the same time so we chunk the requests.